pkg/metallb/bgppeer.go

```diff
@@ -397,8 +421,9 @@ func (builder *BGPPeerBuilder) WithNodeSelector(nodeSelector map[string]string)
 		return builder
 	}
 
-	ndSelector := []mlbtypes.NodeSelector{{MatchLabels: nodeSelector}}
-	builder.Definition.Spec.NodeSelectors = ndSelector
+	builder.Definition.Spec.NodeSelectors = []metav1.LabelSelector{{
```

Comment:
What's the idea of this change?

Reply:
They changed it upstream in mlbtypesv1beta2